src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceService.kt

```diff
@@ -143,11 +150,8 @@ class AttendanceService(
     offenderBooking: OffenderBooking,
     bookingId: Long,
   ) =
-    offenderProgramProfileRepository.findByCourseActivityCourseActivityIdAndOffenderBookingBookingIdAndProgramStatusCode(
-      courseSchedule.courseActivity.courseActivityId,
-      offenderBooking.bookingId,
-      "ALLOC",
-    )
+    offenderProgramProfileRepository.findByCourseActivityAndOffenderBooking(courseSchedule.courseActivity, offenderBooking)
+      .maxByOrNull { it.startDate }
```

Comment:
As we can udpate if allocation ended don't restrict to status ALLOC - just get the latest